src/diffusers/pipelines/step1x_edit/pipeline_step1x_edit.py

```diff
@@ -0,0 +1,1025 @@
+# Copyright 2025 Step1X-Edit Team and The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import inspect
+from typing import Any, Callable, Dict, List, Optional, Union
+
+import numpy as np
+import torch
+import math
+from qwen_vl_utils import process_vision_info
+from transformers import Qwen2_5_VLForConditionalGeneration, Qwen2_5_VLProcessor
+
+from ...image_processor import Image, PipelineImageInput, VaeImageProcessor
+from ...models import AutoencoderKL, Step1XEditTransformer2DModel
+from ...schedulers import FlowMatchEulerDiscreteScheduler
+
+from ...utils import is_torch_xla_available, logging, replace_example_docstring
+from ...utils.torch_utils import randn_tensor
+from ..pipeline_utils import DiffusionPipeline
+from .pipeline_output import Step1XEditPipelineOutput
+
+if is_torch_xla_available():
+    import torch_xla.core.xla_model as xm
+    XLA_AVAILABLE = True
+else:
+    XLA_AVAILABLE = False
+
+logger = logging.get_logger(__name__)  # pylint: disable=invalid-name
+
+EXAMPLE_DOC_STRING = """
+    Examples:
+        ```py
+        >>> import torch
+        >>> from diffusers import Step1XEditPipeline
+        >>> from diffusers.utils import load_image
+
+        >>> pipe = Step1XEditPipeline.from_pretrained("stepfun-ai/Step1X-Edit-v1p1-diffusers", torch_dtype=torch.bfloat16)
+        >>> pipe.to("cuda")
+        >>> image = load_image(
+        ...     "https://github.com/stepfun-ai/Step1X-Edit/blob/main/examples/0000.jpg?raw=true"
+        ... ).convert("RGB")
+        >>> prompt = "Add pendant with a ruby around this girl's neck."
+        
+        >>> image = pipe(
+                image=image,
+                prompt=prompt, 
+                num_inference_steps=28,
+                size_level=1024,
+                guidance_scale=6.0,
+                generator=torch.Generator().manual_seed(1234),
+            ).images[0]
+        >>> image.save("output.png")
+        ```
+"""
+
+QWEN25VL_PREFIX = '''Given a user prompt, generate an "Enhanced prompt" that provides detailed visual descriptions suitable for image generation. Evaluate the level of detail in the user prompt:
```

Comment:
Can this be made a part of the pipeline configuration in the `__init__()` so that users can modify the system prompt or would you not recommend doing that?